modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/util/DataDefinitionUtilTest.java

```diff
@@ -0,0 +1,180 @@
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.util.DDMFormFieldUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Gergely Szalay
+ */
+public class DataDefinitionUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		_fieldName = RandomTestUtil.randomString(8);
+		_fieldReference = RandomTestUtil.randomString();
+	}
+
+	@Test
+	public void testValidateDefinitionFields() {
+		_testValidateDefinitionFieldsWithExistingDataDefinition();
+		_testValidateDefinitionFieldsWithExistingInvalidName();
+		_testValidateDefinitionFieldsWithNewDataDefinition();
+	}
+
+	private DataDefinition _getDataDefinition() {
+		DataDefinition dataDefinition = new DataDefinition();
+
+		dataDefinition.setDefaultDataLayout(_getDataLayout(_fieldName));
+
+		DataDefinitionField dataDefinitionField = new DataDefinitionField();
+
+		dataDefinitionField.setName(_fieldName);
+
+		dataDefinitionField.setCustomProperties(
+			HashMapBuilder.<String, Object>put(
+				"fieldReference", _fieldReference
+			).build());
+
+		dataDefinition.setDataDefinitionFields(
+			new DataDefinitionField[] {dataDefinitionField});
+
+		return dataDefinition;
+	}
+
+	private String _getDataDefinitionFieldName(DataDefinition dataDefinition) {
+		DataDefinitionField dataDefinitionField =
+			dataDefinition.getDataDefinitionFields()[0];
+
+		return dataDefinitionField.getName();
+	}
+
+	private DataLayout _getDataLayout(String fieldName) {
+		DataLayout dataLayout = new DataLayout();
+
+		dataLayout.setDataLayoutPages(
+			new DataLayoutPage[] {
+				new DataLayoutPage() {
+					{
+						dataLayoutRows = new DataLayoutRow[] {
+							new DataLayoutRow() {
+								{
+									dataLayoutColumns = new DataLayoutColumn[] {
+										new DataLayoutColumn() {
+											{
+												fieldNames = new String[] {
+													fieldName
+												};
+											}
+										}
+									};
+								}
+							}
+						};
+						description = HashMapBuilder.<String, Object>put(
+							"en_US", "Description"
+						).build();
+						title = HashMapBuilder.<String, Object>put(
+							"en_US", "Title"
+						).build();
+					}
+				}
+			});
+
+		return dataLayout;
+	}
+
+	private DDMStructure _getDDMStructure(String existingFieldName) {
+		DDMStructure ddmStructure = Mockito.mock(DDMStructure.class);
+		DDMForm ddmForm = Mockito.mock(DDMForm.class);
+
+		Mockito.when(
+			ddmStructure.getDDMForm()
+		).thenReturn(
+			ddmForm
+		);
+
+		DDMFormField ddmFormField = new DDMFormField(existingFieldName, "text");
```

Comment:
@gergelyszalay, based on order (SF), this should be before ddmStructure.getDDMForm 

first ddmForm.getDDMFormFieldsReferencesMap then ddmStructure.getDDMForm